### PR TITLE
n19845 notes and questions

### PR DESCRIPTION
**Pull request by jonatack**
Timely that Tor v2 deprecation is the day before the meeting!

### CODE REVIEW

**Comment by jonatack** [comment]
WIP, need to add some deeper technical notes and harder questions. There's some overlap with the 19031 meeting in July that I've tried/will try to avoid.

**Comment by jonatack** [comment]
Timely that Tor v2 deprecation is the day before the meeting!

File: _posts/2020-09-16-#19845.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  "To very quickly summarize why we are deprecating, in one word: Safety. Onion
+  service v2 uses RSA1024 and 80 bit SHA1 (truncated) addresses [1]. It also
+  still uses the TAP [2] handshake which has been entirely removed from Tor for
+  many years now _except_ v2 services. Its simplistic directory system exposes
+  it to a variety of enumeration and location-prediction attacks that give HSDir
+  relays too much power to enumerate or even block v2 services. Finally, v2
+  services are not being developed nor maintained anymore. Only the most severe
+  security issues are being addressed."
+
+- A [previous review club meeting discussed the parent PR 19031](/19031) and
```
Comment (jonatack) [inline _posts/2020-09-16-#19845.md:46]: removed "and"

**Review by jnewbery — COMMENTED** [review]
These are great notes and questions. Thanks Jon!

Just one comment inline. Let me know what you think.

File: _posts/2020-09-16-#19845.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  now _except_ v2 services. Its simplistic directory system exposes it to a
+  variety of enumeration and location-prediction attacks that give HSDir relays
+  too much power to enumerate or even block v2 services. Finally, v2 services
+  are not being developed nor maintained anymore. Only the most severe security
+  issues are being addressed."
+
+- A [previous review club meeting discussing the parent PR 19031](/19031)
+  provides more background and context.
+
+### Networks newly supported by PR 19845
```
Comment (jonatack) [inline _posts/2020-09-16-#19845.md:49]: suggestion by @vasild here, as support isn't fully operational yet for these:

"New addresses recognized by PR 19845"

File: _posts/2020-09-16-#19845.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+- The `addrv2` feature is enabled by OR-ing a integer flag, `ADDRV2_FORMAT`
+  ([code](https://github.com/bitcoin/bitcoin/pull/19845/files#diff-76d15e11a95be7a4aee1eb89de6098caR32)),
+  into the stream version.
+
+- The PR makes use of the custom `Span` type, which was introduced into Bitcoin
+  Core in [PR 12886](https://github.com/bitcoin/bitcoin/pull/12886) and
+  represents a somewhat vector-like view to a range of contiguous elements in
+  memory analogious to `std::span` in C++20. If you are not familiar with
+  `Span`, it was discussed in a [recent review club meeting](/18468) hosted by
```
[the remark this answers is not in the document]
Comment (jonatack) [reply _posts/2020-09-16-#19845.md:87]: Done in #256.

File: _posts/2020-09-16-#19845.md (hunk 2 above)
Comment (jonatack) [reply _posts/2020-09-16-#19845.md:49]: Done in #256.

**Pull request merged**